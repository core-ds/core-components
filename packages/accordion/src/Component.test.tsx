import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { Accordion, AccordionProps } from '.';

const control = (
    <div
        style={{
            backgroundColor: 'rgba(55, 120, 251, 0.1)',
            border: '1px dashed rgb(55, 120, 251)',
            boxSizing: 'border-box',
            borderRadius: '8px',
            height: '24px',
            width: '24px',
        }}
    />
);

const renderComponent = (props: Partial<AccordionProps> = {}) =>
    render(
        <Accordion header='Зачем нужен аккордион?' {...props}>
            Используется для создания интерактивных списков, которые можно разворачивать и
            сворачивать для отображения дополнительной информации
        </Accordion>,
    );

describe('Accordion', () => {
    describe('Display test', () => {
        it('should match snapshots for base', () => {
            const { container } = renderComponent();

            expect(container).toMatchSnapshot();
        });
    });

    describe('Accordion className test', () => {
        it('should set "dataTestId" prop', () => {
            const dataTestId = 'data-test-id';
            const { getByTestId } = renderComponent({ dataTestId });

            expect(getByTestId(dataTestId).tagName).toBe('DIV');
        });

        it('should set "className" prop', () => {
            const dataTestId = 'data-test-id';
            const className = 'class-name';
            const { getByTestId } = renderComponent({ dataTestId, className });

            expect(getByTestId(dataTestId)).toHaveClass(className);
        });

        it('should set "headerClassName" prop', () => {
            const headerClassName = 'class-name';
            const { getByText } = renderComponent({ headerClassName });

            const header = getByText('Зачем нужен аккордион?').parentElement;

            expect(header).toHaveClass(headerClassName);
        });

        it('should set "controlClassName" prop', () => {
            const controlClassName = 'class-name';
            const { container } = renderComponent({ controlClassName });

            const control = container.querySelector(`.control`);

            expect(control).toHaveClass(controlClassName);
        });

        it('should set "containerClassName" prop', () => {
            const containerClassName = 'class-name';
            const { container } = renderComponent({ containerClassName });

            const body = container.querySelector(`.${containerClassName}`);

            expect(body).toHaveClass(containerClassName);
        });

        it('should set "bodyClassName" prop', () => {
            const bodyClassName = 'class-name';
            const { container } = renderComponent({ bodyClassName });

            const body = container.querySelector(`.${bodyClassName}`);

            expect(body).toHaveClass(bodyClassName);
        });
    });

    describe('Accordion callbacks calls', () => {
        it('should call "onExpandedChange" prop', () => {
            const onExpandedChange = jest.fn();
            const { getByRole } = renderComponent({
                onExpandedChange,
            });
            const control = getByRole('button');

            control.click();

            expect(onExpandedChange).toBeCalledTimes(1);
        });
    });

    describe('Accordion content visibility', () => {
        it('should set "controlPosition" prop', () => {
            const { container } = renderComponent({ controlPosition: 'start' });

            const control = container.querySelector(`.control`);

            expect(control).toHaveClass('startPosition');
        });

        it('should set "expanded" prop', () => {
            const { container, debug } = renderComponent({ expanded: true });

            const expandedClassName = 'expandedBody';
            const body = container.querySelector(`.${expandedClassName}`);

            expect(body).toHaveClass(expandedClassName);
        });
    });
});
