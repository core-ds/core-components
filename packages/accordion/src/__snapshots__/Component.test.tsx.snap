// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Display test should match snapshots for base 1`] = `
<div>
  <div
    class="accordion"
  >
    <div
      aria-expanded="false"
      class="container"
      role="button"
      tabindex="0"
    >
      <div
        class="header"
      >
        <span
          class="primary-large medium"
        >
          Зачем нужен аккордион?
        </span>
      </div>
      <div
        class="control"
      >
        <svg
          class="icon"
          fill="currentColor"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
          />
        </svg>
      </div>
    </div>
    <div
      class="body"
    >
      <div
        class="bodyContent"
      >
        <span
          class="primary-medium"
        >
          Используется для создания интерактивных списков, которые можно разворачивать и сворачивать для отображения дополнительной информации
        </span>
      </div>
    </div>
  </div>
</div>
`;
